GPU/Common/TextureCacheCommon.cpp

```diff
@@ -625,18 +625,15 @@ std::vector<AttachCandidate> TextureCacheCommon::GetFramebufferCandidates(const
 
 	std::vector<AttachCandidate> candidates;
 
-	RasterChannel channel = Memory::IsDepthTexVRAMAddress(entry.addr) ? RasterChannel::RASTER_DEPTH : RasterChannel::RASTER_COLOR;
-	if (channel == RasterChannel::RASTER_DEPTH && !gstate_c.Supports(GPU_SUPPORTS_DEPTH_TEXTURE)) {
-		// Depth texture not supported. Don't try to match it, fall back to the memory behind..
-		return std::vector<AttachCandidate>();
-	}
-
 	const std::vector<VirtualFramebuffer *> &framebuffers = framebufferManager_->Framebuffers();
 
 	for (VirtualFramebuffer *framebuffer : framebuffers) {
 		FramebufferMatchInfo match{};
-		if (MatchFramebuffer(entry, framebuffer, texAddrOffset, channel, &match)) {
-			candidates.push_back(AttachCandidate{ match, entry, framebuffer, channel, channel == RASTER_COLOR ? framebuffer->colorBindSeq : framebuffer->depthBindSeq });
+		if (MatchFramebuffer(entry, framebuffer, texAddrOffset, RASTER_COLOR, &match)) {
+			candidates.push_back(AttachCandidate{ match, entry, framebuffer, RASTER_COLOR, framebuffer->colorBindSeq });
+		}
+		if (gstate_c.Supports(GPU_SUPPORTS_DEPTH_TEXTURE) && MatchFramebuffer(entry, framebuffer, texAddrOffset, RASTER_DEPTH, &match)) {
+			candidates.push_back(AttachCandidate{ match, entry, framebuffer, RASTER_DEPTH, framebuffer->depthBindSeq });
```

Comment:
I think we might want to swap these lines:
https://github.com/hrydgard/ppsspp/blob/6cc8ca517fd5f7550873a14e431a500ad3d40766/GPU/Common/FramebufferManagerCommon.cpp#L524

I'd prefer to give color slight preference if they were bound right next to each other (I realize you checked for the overlap case, but I think it's a general thing.)

Also, I wonder if we should be trying to bind a null texture on `!gstate_c.Supports(GPU_SUPPORTS_DEPTH_TEXTURE)` rather than going to RAM... it feels potentially more consistent.  Not sure.

-[Unknown]

Reply:
Those lines will be separated (and in your preferred order) in #15858  :)

Yes, binding a null texture makes more sense, I'll do that.